syweb/webclient/app/settings/settings-controller.js

```diff
@@ -455,4 +458,50 @@ function($scope, matrixService, modelService, eventHandlerService, notificationS
             dialogService.showError(err);
         });
     };
+
+    $scope.passwordValid = function() {
+        if ($scope.password.oldpw == '') {
+            return false;
+        }
+        if ($scope.password.newpw == '') {
+            return false;
+        }
+        if ($scope.password.confirmnewpw == '') {
+            return false;
+        }
+        return true;
+    };
+
+    $scope.changePassword = function() {
+        if ($scope.password.newpw != $scope.password.confirmnewpw) {
```

Comment:
This check feels as if it should be above in `$scope.passwordValid`. This would make the button be enabled only if the password match, which is nicer than having to click the button then be fonxed.